article/app/model/dotcomponents/DotcomponentsDataModel.scala

```diff
@@ -0,0 +1,196 @@
+    isImmersive: Boolean,
+    page: PageData,
+    nav: NavMenu
+)
+
+case class ContentFields(
+    standfirst: Option[String],
+    main: String,
+    body: String, 
+    blocks: Blocks
+)
+
+case class DotcomponentsDataModel(
+    contentFields: ContentFields,
+    config: Config,
+    tags: List[Tag],
+    version: Int
+)
+
+object Block {
+  implicit val blockElementWrites: Writes[BlockElement] = Json.writes[BlockElement]
+  implicit val writes = Json.writes[Block]
+}
+
+object Blocks {
+  implicit val writes = Json.writes[Blocks]
+}
+
+object ContentFields {
+  implicit val writes = Json.writes[ContentFields]
+}
+
+object TagProperties {
+  implicit val writes = Json.writes[TagProperties]
+}
+
+object Tag {
+  implicit val writes = Json.writes[Tag]
+}
+
+object PageData {
+  implicit val writes = Json.writes[PageData]
+}
+
+object Config {
+  implicit val writes = Json.writes[Config]
+}
+
+object DotcomponentsDataModel {
+
+  val VERSION = 1
+
+  def fromArticle(articlePage: ArticlePage, request: RequestHeader): DotcomponentsDataModel = {
+
+    val article = articlePage.article
+
+    val blocks: List[Block] = article.blocks match {
+      case Some(bs) => bs.body.map(bb => Block(bb.bodyHtml, bb.elements.toList)).toList
+      case None => List()
+    }
+
+    val dcBlocks = Blocks(blocks)
+
+    val contentFields = ContentFields(
+      article.fields.standfirst,
+      article.fields.main,
+      article.fields.body,
+      dcBlocks
+    )
+
+    val jsConfig = (k: String) => articlePage.getJavascriptConfig.get(k).map(_.as[String])
+
+    val pageData = PageData(
+      article.tags.contributors.map(_.name).mkString(","),
+      article.metadata.id,
+      article.metadata.pillar.map(_.toString),
+      Configuration.ajax.url,
+      article.trail.webPublicationDate.getMillis,
+      article.metadata.section.map(_.value),
+      article.trail.headline,
+      article.metadata.webTitle,
+      article.trail.byline.getOrElse(""),
+      jsConfig("contentId"),   // source: content.scala
+      jsConfig("authorIds"),   // source: meta.scala
+      jsConfig("keywordIds"),  // source: tags.scala and meta.scala
+      jsConfig("toneIds"),     // source: meta.scala
+      jsConfig("seriesId"),    // source: content.scala
+      article.metadata.isHosted,
+      Configuration.debug.beaconUrl
+    )
+
+    val tags = article.tags.tags.map(
+      t => Tag(
+        TagProperties(
+          t.id,
+          t.properties.tagType,
+          t.properties.webTitle,
+          t.properties.twitterHandle
+        )
+      )
+    )
+
+    val navMenu = NavMenu(articlePage, Edition(request))
+
+    val config = Config(
+      article.isImmersive,
+      pageData,
+      navMenu
+    )
+
+    DotcomponentsDataModel(
+      contentFields,
+      config,
+      tags,
+      VERSION
+    )
+
+  }
+
+  def toJsonString(model: DotcomponentsDataModel): String = {
+
+    // make what we have look a bit closer to what dotcomponents currently expects
```

Comment:
Is this meant to be permanent or a stepping stone to a more radical change in the model?

As I'm not convinced we should seek to preserve the shape of the existing model. It seems better to define it in whatever way is most convenient for rendering to consume. 

(As an example of where this doesn't seem to be the case, why is `isImmersive` at the top-level of the config object along with 'page' and 'nav' instead of within page?)

Reply:
Yeah this is just the first step to cut down the existing data model so the potential for making breaking changes is reduced/ more efficient / one place to look at while we decide what the want the model to look at.

Revisions to change the structure will be under future PRs